File: final_dataset_A/generated_code/chatgpt/python/file_upload_endpoint_detailed-chatgpt.py
from flask import Flask, request, jsonify
import os

app = Flask(__name__)

UPLOAD_FOLDER = "uploads"
os.makedirs(UPLOAD_FOLDER, exist_ok=True)

@app.route("/upload", methods=["POST"])
def upload_files():
    # Check if files are provided
    if "files" not in request.files:
        return jsonify({
            "success": False,
            "message": "No files provided."
        }), 400

    files = request.files.getlist("files")
    uploaded_files = []

    # Store uploaded files
    for file in files:
        if file.filename == "":
            continue

        file_path = os.path.join(UPLOAD_FOLDER, file.filename)
        file.save(file_path)

        uploaded_files.append({
            "filename": file.filename,
            "path": file_path,
            "size": os.path.getsize(file_path)
        })

    # JSON response
    return jsonify({
        "success": True,
        "message": "Files uploaded successfully.",
        "count": len(uploaded_files),
        "files": uploaded_files
    }), 200

if __name__ == "__main__":
    app.run(debug=True)

File: final_dataset_A/generated_code/chatgpt/java/FileUploadEndpointDetailed-chatgpt.java
package com.example.demo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.nio.file.*;
import java.util.*;

@RestController
@RequestMapping("/api/files")
public class FileUploadController {

    private static final String UPLOAD_DIR = "uploads";

    @PostMapping("/upload")
    public ResponseEntity<Map<String, Object>> uploadFiles(
            @RequestParam("files") MultipartFile[] files) {

        Map<String, Object> response = new HashMap<>();
        List<Map<String, Object>> uploadedFiles = new ArrayList<>();

        try {
            Path uploadPath = Paths.get(UPLOAD_DIR);

            if (!Files.exists(uploadPath)) {
                Files.createDirectories(uploadPath);
            }

            for (MultipartFile file : files) {
                if (file.isEmpty()) {
                    continue;
                }

                Path filePath = uploadPath.resolve(file.getOriginalFilename());
                Files.write(filePath, file.getBytes());

                Map<String, Object> fileInfo = new HashMap<>();
                fileInfo.put("fileName", file.getOriginalFilename());
                fileInfo.put("size", file.getSize());
                fileInfo.put("contentType", file.getContentType());
                fileInfo.put("path", filePath.toString());

                uploadedFiles.add(fileInfo);
            }

            response.put("success", true);
            response.put("message", "Files uploaded successfully.");
            response.put("count", uploadedFiles.size());
            response.put("files", uploadedFiles);

            return ResponseEntity.ok(response);

        } catch (IOException e) {
            response.put("success", false);
            response.put("message", "File upload failed.");
            response.put("error", e.getMessage());

            return ResponseEntity.internalServerError().body(response);
        }
    }
}